Assets/Scripts/Terrain/StreamingWorld.cs

```diff
@@ -773,7 +774,8 @@ private IEnumerator UpdateLocation(int index, bool allowYield)
                             go.hideFlags = defaultHideFlags;
                             go.transform.parent = locationObject.transform;
                             go.transform.localPosition = blockOrigin;
-                            dfLocation.ApplyClimateSettings();
+                            if (allowYield)
+                                dfLocation.ApplyClimateSettings();
```

Comment:
I assumed that it's important to apply climate settings before yielding for the next frame.. but I don't really know if it makes a difference.

Reply:
The terrain and locations are loaded in around player when they're exploring outside world, and I think my reasoning was that player might see texture swaps happen (like pop-in) if climate settings were delayed all the way to end. It's been a while though, some of this stuff dates back to 2014 when it was just DFTFU. It's worth testing without though under a variety of conditions to see if any texture swap pop-in is visible. This might even turn out to be unnecessary. :)